docs/dg/dev/guidelines/performance-guidelines/database-performance-quidelines.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Performance Guidance -  Batch Processing Traits in Propel ORM
+description: Learn how to optimize Propel ORM performance using ActiveRecordBatchProcessorTrait and CascadeActiveRecordBatchProcessorTrait for efficient batch processing, reduced database load, and support for complex entity relationships. Includes usage examples and best practices.
+last_updated: Jun 20, 2025
+template: concept-topic-template
+related:
+  - title: General performance guidelines
+    link: docs/dg/dev/guidelines/performance-guidelines/general-performance-guidelines.html
+---
+
+## Performance Guidance: Batch Processing Traits in Propel ORM
+
+### Prerequirements
+Update your Propel package to the: [spryker/propel:^3.45.0](https://github.com/spryker/propel/releases/tag/3.45.0)
+
+### ActiveRecordBatchProcessorTrait
+- This trait provides performance optimizations for database operations by allowing you to save Propel entities in batches.
+- It enables saving multiple records in a single operation, helping to resolve the - https://docs.spryker.com/docs/dg/dev/troubleshooting/troubleshooting-performance-issues/n+1-problem.html
+- It reduces database round trips and improves overall performance.
+
+#### Basic Usage Example:
```

Comment:
Please explain for the below example:
1. What BatchProcessor is covering: "findOrCreateEntities", "persist"  and in what way (also next p.) and what exactly does "commit" do?  
2. What exactly will happen on the DB level - what kind of queries are generated? Possible ideas: It concats multiple inserts into one transaction; It creates batch query INSERT INTO .. VALUES (), (), () ...

Reply:
sorry, but i found this comments not relevant
this is example to show usage, do not explain how it works under the hood

why it's important - all the documentation should explain usage (as for me), if we will try to explain all detail under the hood, it would take more space than a code itself